src/main/java/io/nats/client/impl/NatsConnection.java

```diff
@@ -224,7 +225,7 @@ else if (cur.equals(first)) {
                 connectError.set(""); // new on each attempt
 
                 timeTraceLogger.trace("setting status to connecting");
-                updateStatus(Status.CONNECTING);
+                updateStatus(Status.CONNECTING, resolved, cur);

```

Comment:
cur is the NatsUri from the server pool. resolved is either cur or a host resolved to an ip address